chromium/allocator.go

```diff
@@ -109,7 +166,40 @@ func (a *Allocator) buildCmdArgs(userDataDir *string, removeDir *bool) ([]string
 	return args, nil
 }
 
-func (a *Allocator) findExecPath() {
+// parseWebsocketURL grabs the websocket address from chrome's output and returns it.
+func (a *Allocator) parseWebsocketURL(ctx context.Context, rc io.Reader) (wsURL string, _ error) {
```

Comment:
Nice job cleaning this up! :clap: We can add integration tests for it now, or even unit tests without launching a browser process. It might make sense to do that here as well, just like you did for `makeUserDataDir`, but it's fine if you want to leave it for later.

Reply:
Sure, it seems easy. I'll take a look.